### PR TITLE
Run all tests

### LINKED ISSUE #137: Run tests for an entire Open-CE environment file
Create script that will run tests for an entire Open-CE environment file.

For each feedstock listed in an Open-CE environment file:
    Look to see if a test file exists:
        If it does, run the test for each provided conda environment file.

### PR DESCRIPTION
**Pull request by bnemanich**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [x] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes #137 .

This PR will allow for tests to be run for every repository within a provided Open-CE environment. All tests will be run within each conda environment that is generated by the build_env command.

A little bit of code cleanup was also added. This included renaming a few variables/functions and moving some chunks of code to their own function. This helped avoid some pylint errors complaining about too many local variables.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: open-ce/build_env.py [hunk 1]
```diff
@@ -121,9 +145,7 @@ def build_env(arg_strings=None):
 
     # Generate conda environment files
     conda_env_files = build_tree.write_conda_env_files(channels=args.channels_list,
```
Comment (bnemanich) [inline open-ce/build_env.py:147]:
```suggestion
    conda_env_files = build_tree.write_conda_env_files(channels=args.channels_list,
                                                                                       output_folder=os.path.abspath(args.output_folder),
                                                                                       path=os.path.abspath(args.output_folder))
```

File: open-ce/conda_env_file_generator.py [hunk 2]
```diff
@@ -43,7 +43,6 @@ def write_conda_env_file(self,
         It returns a list of paths to the files that were writeen.
```
Comment (bethune-bryant) [inline open-ce/conda_env_file_generator.py:43]: Bit of  a nit, but this still says it returns a list.
```suggestion
        It returns the path to the file that was written.
```

**Review by bethune-bryant — APPROVED** [review]
Lgtm

File: open-ce/build_env.py [hunk 3]
```diff
@@ -68,6 +69,29 @@ def make_parser():
...
+        build_tree (BuildTree): The build tree containing the tests
+        conda_env_files (dict): A dictionary where the key is a variant string and the value
+                                is the name of a conda environment file.
+    """
+    failed_tests = []
+    # Run test commands for each conda environment that was generated
+    for variant_string, conda_env_file in conda_env_files.items():
+        test_commands = build_tree.get_test_commands(variant_string)
+        if test_commands:
+            print("\n*** Running tests within the " + conda_env_file + " conda environment ***\n")
```
Comment (npanpaliya) [inline open-ce/build_env.py:86]: Is `conda_env_file` absolute path of the conda env file? If yes, then this statement probably needs correction.

File: open-ce/build_tree.py [hunk 4]
```diff
@@ -91,11 +92,10 @@ def _make_hash(to_hash):
     '''Generic hash function.'''
     return hash(str(to_hash))
 
-def _create_recipes(repository, recipes, variant_config_files, variants, channels):#pylint: disable=too-many-locals
+def _create_commands(repository, recipes, variant_config_files, variants, channels):#pylint: disable=too-many-locals
     """
-    Create a recipe dictionary for each recipe within a repository. The dictionary
-    will have all of the information needed to build the recipe, as well as to
-    create the dependency tree.
+    Create a BuildCommand for each recipe within a repository.
```
Comment (npanpaliya) [inline open-ce/build_tree.py:97]: Should it be `Create a list of BuildCommands` ?

File: open-ce/build_tree.py [hunk 5]
```diff
@@ -106,28 +106,30 @@ def _create_recipes(repository, recipes, variant_config_files, variants, channel
...
+                                           build_type=variants['build_type'],
+                                           mpi_type=variants['mpi_type'],
+                                           cudatoolkit=variants['cudatoolkit'],
+                                           run_dependencies=run_deps,
+                                           host_dependencies=host_deps,
+                                           build_dependencies=build_deps,
+                                           test_dependencies=test_deps,
+                                           channels=channels if channels else []))
+
+    test_commands = test_feedstock.gen_test_commands()
```
Comment (npanpaliya) [inline open-ce/build_tree.py:129]: gen_test_commands() has a argument which is default to tests/open-ce-tests.yaml. I don't see that file anywhere in the repo. Is gen_test_commands() independent of any of the arguments of _create_commands(...)? If yes, then it is generic for all and should not be here. I think I am missing something here.

Comment (npanpaliya) [reply open-ce/build_tree.py:129]: Now I got where is that open-ce-tests.yaml file. It will be added to each feedstock.

**Pull request merged**